test/unittests/utils/test_check_dependency.py

```diff
@@ -0,0 +1,77 @@
+import os
+
+import pytest
+
+
+def test_check_dependency():
+    from clinica.utils.check_dependency import is_binary_present
+
+    assert not is_binary_present("foo")
+    assert is_binary_present("ls")
+
+
+def test_check_environment_variable():
+    from clinica.utils.check_dependency import check_environment_variable
+    from clinica.utils.exceptions import ClinicaMissingDependencyError
+
+    with pytest.raises(
+        ClinicaMissingDependencyError,
+        match="Clinica could not find foo software: the FOO variable is not set.",
+    ):
+        check_environment_variable("FOO", "foo")
+    os.environ["FOO"] = "bar"
+    with pytest.raises(
+        ClinicaMissingDependencyError,
+        match="The FOO environment variable you gave is not a folder",
+    ):
+        check_environment_variable("FOO", "foo")
+    os.environ["FOO"] = "."
+    assert check_environment_variable("FOO", "foo") == "."
+    os.environ.pop("FOO")
+
+
+def test_check_software():
```

Comment:
It feels like this function could easily be factorized in two functions using pytest. One would test the errors, while the other tests the assertions. WDYT ?

Reply:
I proposed a possible refactoring in cdf8f93
Let me know if this is more or less what you had in mind.